tests/src/core/testqgslabelingengine.cpp

```diff
@@ -1274,6 +1274,18 @@ void TestQgsLabelingEngine::testCurvedLabelCorrectLinePlacement()
 
   img = job2.renderedImage();
   QVERIFY( imageCheck( QStringLiteral( "label_curved_label_below_1" ), img, 20 ) );
+
+  // and on line with distance value...
+  settings.lineSettings().setPlacementFlags( QgsLabeling::LinePlacementFlag::OnLine | QgsLabeling::LinePlacementFlag::MapOrientation );
+  settings.dist = 5.0;
```

Comment:
Could we also get a test with negative distance, and some using the parallel mode too?